core/tests/protractor_utils/users.js

```diff
@@ -17,85 +17,51 @@
-  // should handle navigation to the login page.
-  if (manualNavigation) {
-    await driver.get(general.SERVER_URL_PREFIX + general.LOGIN_URL_SUFFIX);
-  }
-  // The statement below uses a browser.wait() to determine if the user has
-  // logged in. Use of waitFor is not possible because the active page is
-  // non-angular.
-  await browser.wait(
-    async() => {
-      try {
-        await driver.findElement(protractor.By.name('email'));
-      } catch (error) {
-        return false;
-      }
-      return true;
-    }, waitFor.DEFAULT_WAIT_TIME_MSECS, 'Login takes too long.');
-  await (await driver.findElement(protractor.By.name('email'))).clear();
-  await (await driver.findElement(protractor.By.name('email'))).sendKeys(email);
+var _createFirebaseAccount = async function(email, isSuperAdmin = false) {
+  var user = await FirebaseAdmin.auth().createUser({
+    email: email,
+    emailVerified: true,
+    password: await HashWasm.md5(email),
+  });
   if (isSuperAdmin) {
-    await (await driver.findElement(protractor.By.name('admin'))).click();
-    let adminCheckboxStatus = await driver.findElement(
-      protractor.By.name('admin')).getAttribute('checked');
-    expect(adminCheckboxStatus).toBeTruthy();
-  }
-  await (await driver.findElement(protractor.By.id('submit-login'))).click();
-  if (manualNavigation) {
-    // The statement below uses a browser.wait() to determine if the user has
-    // logged in. Use of waitFor is not possible because the active page is
-    // non-angular.
-    await browser.wait(
-      async() => {
-        let loginStatusHeaderElement = (
-          await driver.findElement(protractor.By.tagName('h3')));
-        let text = await loginStatusHeaderElement.getText();
-        return text !== 'Logged In';
-      }, waitFor.DEFAULT_WAIT_TIME_MSECS, 'Login takes too long.');
+    await FirebaseAdmin.auth().setCustomUserClaims(
+      user.uid, {role: 'super_admin'});
   }
 };
 
-var login = async function(email, manualNavigation = true) {
-  await _login(email, false, manualNavigation);
+var login = async function(email) {
+  // Use of element and action is not possible because the login page is
+  // non-angular. The full url is also necessary.
+  await browser.driver.get(
+    general.SERVER_URL_PREFIX + general.LOGIN_URL_SUFFIX);
+
+  await waitFor.alertToBePresent();
+  const alert = await browser.switchTo().alert();
+  await alert.sendKeys(email);
+  await alert.accept();
+
+  await waitFor.pageToFullyLoad();
 };
 
 var logout = async function() {
-  // Use of action is not possible because logout page is non-angular.
-  var driver = browser.driver;
-  await driver.get(general.SERVER_URL_PREFIX + general.LOGIN_URL_SUFFIX);
-  await (await driver.findElement(protractor.By.id('submit-logout'))).click();
+  await browser.driver.get(general.SERVER_URL_PREFIX + '/');
+
+  await general.openProfileDropdown();
+  await action.click('Logout link', $('.protractor-test-logout-link'));
```

Comment:
Shouldn't the second argument be defined like element.by.css(), just like all the other e2e tests?

Reply:
Done.